### PR TITLE
[azquery] regenerate and clean up

### PR DESCRIPTION
**Pull request by gracewilcox**
* regenerate code with latest version of code generator
* update autorest.md- separating the transforms for the different clients
* consolidate utils_test.go
* update dependencies

### CODE REVIEW

File: sdk/monitor/azquery/autorest.md [hunk 1]
```diff
@@ -1,25 +1,19 @@
...
-    - https://github.com/Azure/azure-rest-api-specs/blob/dba6ed1f03bda88ac6884c0a883246446cc72495/specification/monitor/resource-manager/Microsoft.Insights/stable/2018-01-01/metrics_API.json
-    - https://github.com/Azure/azure-rest-api-specs/blob/dba6ed1f03bda88ac6884c0a883246446cc72495/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-12-01-preview/metricNamespaces_API.json
 license-header: MICROSOFT_MIT_NO_VERSION
 module: github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery
 openapi-type: "data-plane"
 output-folder: ../azquery
 override-client-name: LogsClient
 security: "AADToken"
-use: "@autorest/go@4.0.0-preview.46"
+use: "@autorest/go@4.0.0-preview.59"
```
Comment (chlowell) [inline sdk/monitor/azquery/autorest.md:16]:
```suggestion
use: "@autorest/go@4.0.0-preview.60"
```

File: sdk/monitor/azquery/utils_test.go [hunk 2]
```diff
@@ -101,20 +80,10 @@ func run(m *testing.M) int {
...
-			panic(err)
-		}
-		err = recording.AddGeneralRegexSanitizer(fakeResourceURI, resourceURI, nil)
-		if err != nil {
-			panic(err)
-		}
-	}
+	workspaceID = getEnvVar("WORKSPACE_ID", fakeWorkspaceID)
+	workspaceID2 = getEnvVar("WORKSPACE_ID2", fakeWorkspaceID2)
+	resourceURI = getEnvVar("WORKSPACE_ID", fakeResourceURI)
```
Comment (chlowell) [inline sdk/monitor/azquery/utils_test.go:85]:
```suggestion
	resourceURI = getEnvVar("RESOURCE_URI", fakeResourceURI)
```

File: sdk/monitor/azquery/utils_test.go [hunk 3]
```diff
@@ -164,12 +133,25 @@ func startMetricsTest(t *testing.T) *azquery.MetricsClient {
 	return client
 }
 
-func lookupEnvVar(s string) string {
-	ret, ok := os.LookupEnv(s)
-	if !ok {
-		panic(fmt.Sprintf("Could not find env var: '%s'", s))
+func getEnvVar(lookupValue string, fakeValue string) string {
+	// get value
+	envVar := fakeValue
```
Comment (chlowell) [inline sdk/monitor/azquery/utils_test.go:138]: Nitpicking, I'd change a couple names here to make clear that we're looking up the value of an environment variable and returning either that value or a fake one
```suggestion
func getEnvVar(envVar string, fakeValue string) string {
	// get value
	value := fakeValue
```

File: sdk/monitor/azquery/autorest.md [hunk 4]
```diff
@@ -1,25 +1,19 @@
 ## Go
 
 ``` yaml
-title: MonitorQueryClient
-description: Azure Monitor Query Go Client
-generated-metadata: false
-
+title: Logs Query Client
```
Comment (chlowell) [inline sdk/monitor/azquery/autorest.md:4]: autorest help claims `title` overrides the client name but actually it doesn't appear to have any effect and we have `override-client-name` as well, which also appears to do nothing. 🤔 should we just omit this?

**Review by chlowell — APPROVED** [review]

File: sdk/monitor/azquery/utils_test.go (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/utils_test.go]: Good catch, thanks! Fixed in [d0c65c0](https://github.com/Azure/azure-sdk-for-go/pull/21965/commits/d0c65c005478be252e03a390bdc8d4ded5ffe04a)

File: sdk/monitor/azquery/utils_test.go (hunk 3 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/utils_test.go]: Fixed in [d0c65c0](https://github.com/Azure/azure-sdk-for-go/pull/21965/commits/d0c65c005478be252e03a390bdc8d4ded5ffe04a)

File: sdk/monitor/azquery/autorest.md (hunk 4 above)
Comment (gracewilcox) [reply sdk/monitor/azquery/autorest.md:4]: `override-client-name` doesn't do anything, so removed in [d0c65c0](https://github.com/Azure/azure-sdk-for-go/pull/21965/commits/d0c65c005478be252e03a390bdc8d4ded5ffe04a)

**Pull request merged**